### ISSUE TITLE
RecommendByAuthorPlugin work only for primary locale

### ISSUE THREAD

**Issue by Vitaliy-1** [pkp/pkp-lib#2568]
Basically, plugin show articles in my case only for primary locale (English) and it shows nothing for secondary (Ukrainian).

**Comment by bozana** [comment]
@Vitaliy-1, could you please explain it a little bit more? Are there maybe also links where I could see it? -- Unfortunately, in my local installation (master branch) I cannot reproduce it :-\

**Comment by Vitaliy-1** [comment]
Hi @bozana,
I have activated the plugin on our site. It is displayed on article_detail page under `Detail` tab: https://e-medjournal.com/index.php/psp/article/view/19
Maybe it is because a conflict between plugins, that use Footer hook. But I actually don't know, what plugin uses it too.

**Comment by ajnyga** [comment]
I have not been able to reproduce the conflict situation with embedGalley and recommendByAuthorPlugin with the latest versions of these plugins and the fix @bozana did a while ago. But it could just be that there are other variables at stake here.

**Comment by bozana** [comment]
@Vitaliy-1, OK, I think I know where is the problem: the authors are the same if their first, last and middle names are the same, as well as country and affiliation (that is multilingual i.e. it has to be the same for all journal languages), s. https://github.com/pkp/ojs/blob/master/plugins/generic/recommendByAuthor/RecommendByAuthorPlugin.inc.php#L84. Thus, I think that the affiliation of your authors for Ukrainian are not the same for all articles :-\ It is a little bit inconvenient, but there is currently no better way we can proof if it is one and the same author :-(

**Comment by ajnyga** [comment]
that's bad. What if the author's affiliation changes? That will happen a lot.

**Comment by bozana** [comment]
Yes, that was always the problem, also in 2.4.x. 
@asmecher, is there any other possibility for now to see if two authors are one and the same author? Could we ignore the affiliation?
:-\

**Comment by Vitaliy-1** [comment]
Yep, that's the case.
Hmm. Our authors are often affiliated to 2 institution and can change, for example, their order from article to article. The only way for the plugin to work not to allow this...

**Comment by asmecher** [comment]
Any name-based disambiguation is destined to fail on a lot of cases. It's never going to be reliable. We did it in OJS 2.x and there's probably still a remnant of that in OJS 3.x, but it shouldn't be relied on or maintained further. Going forward, it's better to use ORCID or a similar disambiguating ID.

**Comment by bozana** [comment]
@asmecher, would that mean that we should change the plugin to look only after the ORCID? Or, in case when ORCID is not there to keep using the old/current method i.e. if first, middle, last name, country and affiliation is the same?

**Comment by asmecher** [comment]
I haven't seen any installations make sufficient use of ORCIDs to really have them take over as the only disambiguation method, unfortunately. One option would be to "force" adoption by removing the fallback on names, but I suppose it would be better citizenship for us to preserve the name-based approach as the fallback...

**Comment by ajnyga** [comment]
Hi all,

I also think that it will be long before we can count on ORCIDs in matters like this. Actually I think that in the case of this plugin such time will never arrive, because there are always old archives with authors with no ORCIDs. So this is definitely not something where we can use "force" adoption. 

The situation may be different, if we are talking about user accounts in OJS. But even here, I know that some older researchers are not going to get an ORCID.

So how about chancing RecommendByAuthorPlugin so that there are **two sets of recommendations**:
1. High value matches: based on ORCID and/or email account.
2. Low lever matches: based on first name + last name (and only that!)

This way we can show with some visual tricks that the case 1 matches are the better ones (guaranteed match), but you may want to check case 2 matches as well.

**Comment by asmecher** [comment]
@ajnyga, I agree with your suggestion.

**Comment by NateWr** [comment]
Closing this as the original report is not a bug. If there is still an enhancement to implement for the plugin, please file a new issue with a more detailed feature request.
